File: iosApp/iosApp/Presentation/Community/PostEditAction.swift
//
//  PostEditAction.swift
//  ConCafe
//
//  Created by 홍희표 on 4/28/26.
//

import Foundation

enum PostEditAction {
    case clickBack
    case changeTitle(String)
    case changeContent(String)
    case clickAddImage
    case addImage(String)
    case removeImage(Int)
    case clickSubmit
    case dismissInfoMessage
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/community/edit/PostEditAction.kt
package com.hhp227.concafe.presentation.main.community.edit

sealed interface PostEditAction {
    data object ClickBack : PostEditAction
    data class ChangeTitle(val value: String) : PostEditAction
    data class ChangeContent(val value: String) : PostEditAction
    data class AddImage(val imageUrl: String) : PostEditAction
    data class RemoveImage(val index: Int) : PostEditAction
    data object ClickAddImage : PostEditAction
    data object ClickSubmit : PostEditAction
    data object DismissInfoMessage : PostEditAction
}
